geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -312,6 +320,23 @@ public String toString() {
     return null;
   }
 
+  private void validateScoreIsDouble(byte[] score) {
+    String scoreString = Coder.bytesToString(score).toLowerCase();
+    switch (scoreString) {
+      case "inf":
+      case "infinity":
+      case "+inf":
+      case "+infinity":
+      case "-inf":
+      case "-infinity":
+        return;
+      default:
+        if (!Coder.bytesToString(score).matches("^[-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?$")) {
+          throw new NumberFormatException(ERROR_NOT_A_VALID_FLOAT);
+        }
+    }
+  }
+
```

Comment:
The _intent_ is to just use a regexp to validate the float instead of doing a full conversion at this point. However, based on our in-person discussion yesterday, if we change how we do the 'double sanitizing', we could avoid even having to call this function at this point. Since I suspect the "handle infinity" PR (GEODE-9317) will get merged before this one, I might rebase at that point and rework this section.

Reply:
In light of the plan to change how we filter infinities and such, I've ditched validateScoreIsDouble and replaced it with makeDoubleWhileHandlingInfinity.